dev/devicelab/lib/framework/ios.dart

```diff
@@ -99,3 +101,85 @@ Future<bool> containsBitcode(String pathToBinary) async {
   });
   return !emptyBitcodeMarkerFound;
 }
+
+/// Creates and boots a new simulator, passes the new simulator's identifier to
+/// `testFunction`, then shuts down and deletes simulator.
+Future<void> testWithNewiOSSimulator(
```

Comment:
This was essentially copied from https://github.com/flutter/flutter/blob/023532d9919463404e0ae2a8aa47d07247ad7e65/dev/devicelab/bin/tasks/ios_app_with_extensions_test.dart#L79.  Next PR will convert that test to use this new function.